src/vsDebugServer.ts

```diff
@@ -0,0 +1,137 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+require('source-map-support').install(); // Enable TypeScript stack traces translation
+import 'reflect-metadata';
+
+/**
+ * This script launches vscode-js-debug in server mode for Visual Studio
+ */
+import * as net from 'net';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import { createGlobalContainer } from './ioc';
+import { IDebugSessionLike, ISessionLauncher, Session } from './sessionManager';
+import { getDeferred, IDeferred } from './common/promiseUtil';
+import DapConnection from './dap/connection';
+import { IPseudoAttachConfiguration } from './configuration';
+import { DebugConfiguration } from 'vscode';
+import { ServerSessionManager } from './serverSessionManager';
+import { ITarget } from './targets/targets';
+import { Readable, Writable } from 'stream';
+
+const storagePath = fs.mkdtempSync(path.join(os.tmpdir(), 'vscode-js-debug-'));
+
+class VSDebugSession implements IDebugSessionLike {
+  constructor(
+    public id: string,
+    name: string,
+    private readonly childConnection: Promise<DapConnection>,
+    public readonly configuration: DebugConfiguration,
+  ) {
+    this._name = name;
+  }
+
+  private _name: string;
+  set name(newName: string) {
+    this._name = newName;
+    this.childConnection
```

Comment:
Can we run into any issues because we do not await this operation?

Reply:
We specifically _don't_ want to await this, since it would block init code. We just want to send the process event with the target name once we are initialized.